File: src/main/java/com/timepath/steam/io/bvdf/BVDFListener.java
package com.timepath.steam.io.bvdf;

/**
 * @author TimePath
 */
interface BVDFListener {

    /**
     * Invoked when a value is read
     *
     * @param key
     * @param value
     */
    void emit(String key, Object value);

    /**
     * Invoked when leaving the previous section
     */
    void pop();

    /**
     * Invoked when moving into a nested section
     *
     * @param section
     */
    void push(Object section);
}

File: src/main/kotlin/com/timepath/steam/io/bvdf/BVDFListener.kt
package com.timepath.steam.io.bvdf

interface BVDFListener {

    /**
     * Invoked when a value is read
     */
    public fun emit(key: String, value: Any)

    /**
     * Invoked when leaving the previous section
     */
    public fun pop()

    /**
     * Invoked when moving into a nested section
     */
    public fun push(section: Any)
}
